custom_components/birthdays/__init__.py

```diff
@@ -45,28 +54,44 @@ async def async_setup(hass, config):
     tasks = [asyncio.create_task(device.update_data()) for device in devices]
     await asyncio.wait(tasks)
 
+    _LOGGER.debug(devices)
+
     return True
 
 
 class BirthdayEntity(Entity):
 
-    def __init__(self, name, date_of_birth, icon, hass):
+    def __init__(self, unique_id, name, date_of_birth, icon, attributes, age_at_next_birthday, hass):
         self._name = name
-        self._date_of_birth = date_of_birth
-        self._icon = icon
-        self._age_at_next_birthday = 0
-        self._state = None
-        name_in_entity_id = slugify(name)
-        self.entity_id = 'birthday.{}'.format(name_in_entity_id)
-        self.hass = hass
+
+        if unique_id is not None:
+            self._unique_id = slugify(unique_id)
+        else: 
+            self._unique_id = slugify(name)
+
+        self._state                = None
+        self._icon                 = icon
+        self._date_of_birth        = date_of_birth
+        self._age_at_next_birthday = age_at_next_birthday
```

Comment:
Line 75 can also be removed, since we store the age in the _extra_state_attributes

Reply:
Removed!